prow/github/client.go

```diff
@@ -672,6 +672,28 @@ func (c *client) request(r *request, ret interface{}) (int, error) {
 	return statusCode, nil
 }
 
+// Make a request with retries. If ret is not nil, unmarshal the response body
+// into it. Returns an error if the exit code is not one of the provided codes.
+// Returns githubError if HTTP Status Code is not 2xx.
+func (c *client) requestWithGitHubError(r *request, ret interface{}) (int, githubError, error) {
```

Comment:
@matthyx @alvaroaleman I was thinking of putting this functionality in the original "request" method and change the signature of the calling functions. With that I'll be able to get rid of this method as it is mostly doing whatever was done in the "request" method. 
I wanted to create an independent PR to do these changes as they are not related to this bug fix.
Please let me know your thoughts

Reply:
I don't think its a good idea to add this method or to have methods that have multiple `error` type returns that are mutually exclusive in general. The correct way for something like this would be to return the `githubError` as the last return and then use [`errors.As`](https://golang.org/pkg/errors/#As) in the caller

Reply:
Found out after a bit more digging in `types.go` that there was no need to extend `githubError` or to make any changes in the `request` function. There's already a datatype `AlternativeClientError` which has the fields corresponding to the response we get from Github. That data type is returned by `request` function, so just need to use `errors.As` as suggested above.

That raises an independent question - is there a need to have the data type `githubError` at all?